src/config/builder.rs

```diff
@@ -78,6 +78,14 @@ pub struct ConfigBuilder {
     /// All configured secrets backends.
     #[serde(default)]
     pub secret: IndexMap<ComponentKey, SecretBackends>,
+
+    /// The duration in seconds to wait for graceful shutdown after SIGINT or SIGTERM are received.
+    /// After the duration has passed, Vector will force shutdown. Default value is 60 seconds. If set
+    /// to -1, Vector will never force shutdown. This value can be set using a
+    /// [cli arg](crate::cli::RootOpts::graceful_shutdown_duration).
+    #[serde(default, skip)]
+    #[doc(hidden)]
+    pub graceful_shutdown_duration: i64,
```

Comment:
I think that might be able to be more gracefully (pun intended) accomplished as (I think? a value of zero wouldn't be super useful?) :

```Rust
    pub graceful_shutdown_duration: Option<NonZeroUsize>,
```

Reply:
Thanks for the review! Does Option<Duration> sound good to you instead?